File: Application/Domain/DTOs/LoanRequestDTO.cs
namespace Domain.DTOs;

public class LoanRequestDTO
{
    public string AccountNumber { get; set; }
    public double RemainingAmount { get; set; }
    public double Amount { get; set; }
    public int Duration { get; set; }
    public double InterestRate { get; set; }
    public double MonthlyPayment { get; set; }
    public DateTime? EndDate { get; set; }

}

File: Database/src/main/java/Database/DTOs/LoanRequestDTO.java
package Database.DTOs;

import com.google.protobuf.Timestamp;

public class LoanRequestDTO {
    private String accountId;
    private double remainingAmount;
    private double interestRate;
    private double monthlyPayment;
    private Timestamp endDate;
    private double loanAmount;

    public LoanRequestDTO(String accountId, double remainingAmount, double interestRate, double monthlyPayment, Timestamp endDate, double loanAmount) {
        this.accountId = accountId;
        this.remainingAmount = remainingAmount;
        this.interestRate = interestRate;
        this.monthlyPayment = monthlyPayment;
        this.endDate = endDate;
        this.loanAmount = loanAmount;
    }

    public String getAccountId() {
        return accountId;
    }
    public double getRemainingAmount() {
        return remainingAmount;
    }
    public double getInterestRate() {
        return interestRate;
    }
    public double getMonthlyPayment() {
        return monthlyPayment;
    }
    public Timestamp getEndDate() {
        return endDate;
    }
    public double getLoanAmount() {
        return loanAmount;
    }

}
